sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/metadata/MetadataContext.java

```diff
@@ -0,0 +1,119 @@
+   *
+   * @param metadataContext the new {@link MetadataContext}, e.g. {@linkplain
+   *     DetachedMetadataContext#toMetadataContext() derived} from a {@link
+   *     DetachedMetadataContext}.
+   */
+  static void createContext(DetachedMetadataContext metadataContext) {
+    MetadataContextHolder.set(metadataContext.toMetadataContext());
+  }
+
+  /**
+   * Transfers the current metadata context to the runnable when executed in a new thread.
+   *
+   * @param runnable The runnable to wrap with the current metadata context.
+   * @return The original runnable wrapped with code to transfer the metadata context when executed
+   *     in a new thread.
+   */
+  static Runnable transferMetadataContext(Runnable runnable) {
+    return MetadataContextUtil.transferMetadataContext(runnable);
+  }
+
+  /**
+   * Transfers the current metadata context to the callable when executed in a new thread.
+   *
+   * @param callable The runnable to wrap with the current metadata context.
+   * @return The original callable wrapped with code to transfer the metadata context when executed
+   *     in a new thread.
+   */
+  static <V> Callable<V> transferMetadataContext(Callable<V> callable) {
+    return MetadataContextUtil.transferMetadataContext(callable);
+  }
+
+  /**
+   * @return all available keys in the metadata context.
+   */
+  Set<String> metadataKeys();
+
+  /**
+   * {@link #valuesByConfiguredMetadataKey(String)} should be preferred to be independent of
+   * different environments.
+   *
+   * @param key a key in the metadata context
+   * @return the values stored in the metadata context by this key
+   */
+  List<String> valuesByMetadataKey(String key);
+
+  /**
+   * {@linkplain #valuesByMetadataKey(String) reads} the values of the metadata context from a key
+   * that is configurable by {@linkplain System#getProperty(String) system properties} or
+   * {@linkplain System#getenv(String) environment variables}. This is the preferred approach to get
+   * information from the metadata context for a specific need.
+   *
+   * <p>The {@code environmentOrPropertyName} should refer to the business use of the context values
+   * in a specific service. The actual key is dependent on the environment where the service is
+   * used. Other services may need the same key in a different context and operators may have their
+   * own assumptions about the naming.
+   *
+   * @param environmentOrPropertyName the name of a {@linkplain System#getProperty(String) system
+   *     property} or {@linkplain System#getenv(String) environment variable} that defines the
+   *     actual key for a specific need of the service.
+   * @return the values stored in the metadata context by this key
+   * @throws KeyConfigurationMissingException if the given {@code environmentOrPropertyName} does
+   *     not resolve to a metadata context key
+   */
+  default List<String> valuesByConfiguredMetadataKey(String environmentOrPropertyName)
```

Comment:
I'd prefer `valuesByEnvKey` instead of "configured". It would make it clearer IMHO that where we look up the key. I guess you don't mean that the value comes from the configuration file.

Reply:
That would match the main production use case. We are also looking it up in the system properties (mainly to be able to test). As the parameter name makes that clear as well, I'm fine with renaming. What do you think about `valuesByKeyFromEnvironment(String environmentOrPropertyName) `. Imho that is more exact.

Reply:
I like `...ByKeyFromEnvironment` 👍